.githooks/pre-commit/test

```diff
@@ -1,9 +1,10 @@
 #!/usr/bin/env node
 
-var spawn = require('child_process').spawn;
-var exec = require('child_process').execSync;
-var gitCachedFiles = exec('git diff --cached --name-only --diff-filter=ACMR').toString().trim();
+var CP = require('child_process');
+var gitCachedFiles = CP.execSync('git diff --cached --name-only --diff-filter=ACMR').toString().trim();
+var opts = {stdio: 'inherit'};
 
 if (gitCachedFiles) {
-    spawn('npm', ['test'], {stdio: 'inherit'}).on('close', process.exit.bind(process));
+    (process.platform === 'win32' ?
+        CP.exec('npm test', opts) : CP.spawn('npm', ['test'], opts)).on('close', process.exit.bind(process));
```

Comment:
Using `execSync` for  executing `npm test` is a bad idea because the script can reach buffer limit. Moreover, `execSync` doesn't proxy any output during execution; so, a user watches nothing but an empty screen. It's a bad UX.